### PR TITLE
 Add generic ECPoint data class (#73)

### PR DESCRIPTION
**Pull request by iaik-jheher**
rebased version of #73, new PR since i moved it out of the fork to have CI tests

### CODE REVIEW

File: datatypes/src/commonMain/kotlin/at/asitplus/crypto/datatypes/CryptoPublicKey.kt [hunk 1]
```diff
@@ -327,18 +343,16 @@ sealed class CryptoPublicKey : Asn1Encodable<Asn1Sequence>, Identifiable {
         fun toAnsiX963Encoded(): ByteArray =
             when (useCompressedRepresentation) {
                 true -> {
-                    val prefix = (2U + compressY().toUInt()).toByte()
-                        .also { require(it == ANSI_COMPRESSED_PREFIX_1 || it == ANSI_COMPRESSED_PREFIX_2) }
                     byteArrayOf(
-                        prefix,
-                        *x.ensureSize(curve.coordinateLengthBytes)
+                        ANSIECPrefix.forSign(yCompressed).prefixByte,
```
Comment (iaik-jheher) [inline datatypes/src/commonMain/kotlin/at/asitplus/crypto/datatypes/CryptoPublicKey.kt:347]: here's a use that would _definitely_ be prettier without `prefixByte`

**Review by iaik-jheher — COMMENTED** [review]
@JesusMcCloud re-tagging you in the new PR

**Comment by JesusMcCloud** [comment]
@iaik-jheher my last commit is the best i could do, since enums impose some limitations. Fell free to accept as much or as little as you like

**Comment by iaik-jheher** [comment]
> @iaik-jheher my last commit is the best i could do, since enums impose some limitations. Fell free to accept as much or as little as you like

lgtm

**Comment by iaik-jheher** [comment]
For reference, in addition to the actual EC point stuff, there are two refactors that caused a few LoCs of change:

* Replaced the `Signum = Boolean` typedef with BigInt's `Sign`
* Refactored the `ANSI_*` constants to be an enum class

Since they overlap on quite some lines I'd prefer not to have to split them apart. Sorry for the mess.

File: datatypes/src/commonMain/kotlin/at/asitplus/crypto/datatypes/ECPoint.kt
Comment (JesusMcCloud) [inline datatypes/src/commonMain/kotlin/at/asitplus/crypto/datatypes/ECPoint.kt:1]: please add some kdoc comment explaining what a generalecpoint and an ecpoint is

**Review by JesusMcCloud — APPROVED** [review]
aweseome work except for misssing kdoc as commented. I'm still approving it right away, so you can just rebase once commented.

**Review by JesusMcCloud — APPROVED** [review]
ship it!

**Pull request merged**